bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/utils/AudioSinkUtils.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.audio.utils;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.audio.AudioSink;
+import org.openhab.core.library.types.PercentType;
+import org.slf4j.Logger;
+
+/**
+ * Some utility methods for sink
+ *
+ * @author Gwendal Roulleau - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class AudioSinkUtils {
+
+    /**
+     * Handle a volume command change and returns a Runnable to restore it.
+     *
+     * @param volume The volume to set
+     * @param sink The sink to set the volume to
+     * @param logger to log error to
+     * @return A runnable to restore the volume to its previous value, or null if no change is required
+     */
+    public static @Nullable Runnable handleVolumeCommand(@Nullable PercentType volume, AudioSink sink, Logger logger) {
```

Comment:
> Why do you pass a logger here?

openHAB has a SAT rule of non static logger. As I didn't want to change this simple class with static method to a full service/component (I wanted to keep it static), I thought of passing logger instance as an argument. But I now see that static logger are used elsewhere and it was not necessary.
Anyway, I moved handleVolumeCommand in AudioManagerImpl, so it has access to the class logger.

> And wouldn't this whole class unnecessary if .setVolume would return the old volume? Restoring the old volume would then be a one-liner.

The whole point of this method, unlike the previous way of doing, is to allow us to do no nothing if not needed (and mutualize code and exception handling). In the previous implementation, volume was always set back to the old value, even if it's the same value. And for some sinks, it is an operation that can take some non negligible time and/or unneeded side effect)
That said, I could have done otherwise, because a Runnable is not the only way to state a no-op operation. A Nullable or Optional old volume value would have worked. But then it wouldn't be a one liner, as it would be mandatory to test for null. As this method is used only in a couple of place in the same way (with a delayed task in mind), a Runnable seems to me a good choice to make it a one-liner. Is also has the added value of managing exception inside. I added some comment in the method to explain this. But it is not very important, so if you want me to change it, no problem, just tell me to.

Another point to mention:
I had to move the transferAndAnalyzeLength method to a service component, to avoid a dependancy issue with the javazoom library used for parsing mp3.